cli/exchange/node.go

```diff
@@ -465,3 +500,292 @@ func NodeListErrors(org string, credToUse string, node string, long bool) {
 		fmt.Printf("%s\n", jsonBytes)
 	}
 }
+
+func ValidateUserInput(org string, credToUse string, userInput policy.UserInput, exNode ExchangeNodes) (bool, string, error) {
+	msgPrinter := i18n.GetMessagePrinter()
+	warningMessage := ""
+
+	serviceOrg := userInput.ServiceOrgid
+	serviceUrl := userInput.ServiceUrl
+	serviceVersionRange := userInput.ServiceVersionRange
+	serviceArch := userInput.ServiceArch
+
+	msgPrinter.Printf("Start validating userinput %v, %v, %v, %v", serviceOrg, serviceUrl, serviceVersionRange, serviceArch)
+	msgPrinter.Println()
+
+	var errorString string
+	if &userInput.ServiceVersionRange == nil || userInput.ServiceVersionRange == "" {
+		def := "0.0.0"
+		serviceVersionRange = def
+	}
+
+	// Convert the version to a version expression.
+	vExp, err := semanticversion.Version_Expression_Factory(serviceVersionRange)
+	if err != nil {
+		errorString = msgPrinter.Sprintf("versionRange %v cannot be converted to a version expression, error %v", userInput.ServiceVersionRange, err)
+		return false, warningMessage, errors.New(errorString)
+	}
+	serviceVersion := vExp.Get_expression()
+
+	// service exist
+	var sdef *exchange.ServiceDefinition
+
+	searchVersion, err := getSearchVersion(serviceVersion)
+	if err != nil {
+		return false, warningMessage, err
+	}
+
+	sdef, err = getService(cliutils.OrgAndCreds(org, credToUse), serviceOrg, serviceUrl, serviceArch, serviceVersion, searchVersion)
+	if sdef == nil {
+		errorString = msgPrinter.Sprintf("Service does not exist for org: %v, url: %v, version: %v, arch: %v, get error: %v \n", serviceOrg, serviceUrl, serviceVersion, serviceArch, err)
```

Comment:
If the org in the userinput is wrong, the errorString duplicates the text that's in the err variable. Either shorten the message in err, or just return the text in err. Here is the error message, see how the service does not exist is duplicated.
Error: Unable to validate exchange node userInput, error: Service does not exist for org: othere2edev@somecomp.com, url: https://bluehorizon.network/services/testservice, version: [0.0.0,INFINITY), arch: amd64, get error: Service does not exist for org: othere2edev@somecomp.com, url: https://bluehorizon.network/services/testservice, version: [0.0.0,INFINITY), arch: amd64

However, if the version range is too high, the err variable contains nil. You need to make the error reporting more consistent. Here's the error for this one:
Error: Unable to validate exchange node userInput, error: Service does not exist for org: e2edev@somecomp.com, url: https://bluehorizon.network/services/testservice, version: [5.0.0,INFINITY), arch: amd64, get error: <nil>